### PR TITLE
[google_maps_flutter] Add iOS unit and UI tests

### PR DESCRIPTION
**Pull request by jmagman**
Add `RunnerTests` XCTest and `RunnerUITest` XCUITest iOS targets to example app.

- Add unit test that creates the plugin and asserts it's not nil, proving linking is set up correctly.
- Add UI test that navigates around, checks for a maps platform view, and handles the location permission popup.

google_maps_flutter part of https://github.com/flutter/flutter/issues/83357

Passing presubmit tests: https://cirrus-ci.com/task/6672684797919232
```
Test Case '-[GoogleMapsUITests testUserInterface]' passed (11.633 seconds).
Test Suite 'GoogleMapsUITests' passed at 2021-05-25 16:00:22.642.
	 Executed 1 test, with 0 failures (0 unexpected) in 11.633 (11.634) seconds
Test Suite 'RunnerUITests.xctest' passed at 2021-05-25 16:00:22.644.
	 Executed 1 test, with 0 failures (0 unexpected) in 11.633 (11.636) seconds
Test Suite 'All tests' passed at 2021-05-25 16:00:22.645.
	 Executed 1 test, with 0 failures (0 unexpected) in 11.633 (11.639) seconds
```
```
Test Case '-[GoogleMapsTests testPlugin]' passed (0.003 seconds).
Test Suite 'GoogleMapsTests' passed at 2021-05-25 16:00:05.826.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.003 (0.004) seconds
Test Suite 'RunnerTests.xctest' passed at 2021-05-25 16:00:05.827.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.003 (0.005) seconds
Test Suite 'All tests' passed at 2021-05-25 16:00:05.827.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.003 (0.008) seconds
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`. See [plugin_tool format](../script/tool/README.md#format-code))
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/example/ios/Podfile [hunk 1]
```diff
@@ -29,6 +29,9 @@ flutter_ios_podfile_setup
 
 target 'Runner' do
   flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
+  target 'RunnerTests' do
+    inherit! :search_paths
+  end
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/Podfile:34]: `RunnerTests` needs to link on Flutter and plugins.  `RunnerUITests` do not since they interact with the UI blackbox.

File: packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerUITests/GoogleMapsUITests.m [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                          if (![allow waitForExistenceWithTimeout:30.0]) {
+                                            XCTFail(@"Failed due to not able to find Allow button");
+                                          }
+                                          [allow tap];
+                                        }
+                                        return YES;
+                                      }];
+}
+
+- (void)testUserInterface {
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerUITests/GoogleMapsUITests.m:45]: Just a smoke test to prove the UI test is set up correctly.

File: packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerTests/GoogleMapsTests.m [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@import google_maps_flutter;
+@import XCTest;
+
+@interface GoogleMapsTests : XCTestCase
+@end
+
+@implementation GoogleMapsTests
+
+- (void)testPlugin {
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerTests/GoogleMapsTests.m:13]: Smoke test to prove linking on the plugin works.

**Review by stuartmorgan — APPROVED** [review]
Awesome, thanks! LGTM :)
